crates/driver/src/domain/quote.rs

```diff
@@ -258,6 +265,48 @@ impl Tokens {
     }
 }
 
+mod encode {
```

Comment:
Does this module here add much? Seems like the same thing could be achieved with a private function `encode_interaction`.

Reply:
no, it doesn't add anything (but does it cost anything)? I thought we are trying to keep our naming module drive and prefer something like:

`solution::encode::interaction(...)` over `solution_encode_interaction(...)` since the former allows you to selectively import only parts of the path (ie within the `encode` module it makes sense to just call into `liquidity()`, `approve()`, etc.) instead of specifying everything.